beacon_node/network/src/service.rs

```diff
@@ -228,125 +244,145 @@ fn spawn_service<T: BeaconChainTypes>(
+                                        let subnet = &subnet_and_attestation.0;
+                                        let attestation = &subnet_and_attestation.1;
+                                        // checks if we have an aggregator for the slot. If so, we process
+                                        // the attestation
+                                        if service.attestation_service.should_process_attestation(
+                                            &id,
+                                            &source,
+                                            subnet,
+                                            attestation,
+                                        ) {
+                                            let _ = service
+                                                .router_send
+                                                .send(RouterMessage::PubsubMessage(id, source, message))
+                                                .map_err(|_| {
+                                                    debug!(service.log, "Failed to send pubsub message to router");
+                                                });
+                                        } else {
+                                            metrics::inc_counter(&metrics::GOSSIP_UNAGGREGATED_ATTESTATIONS_IGNORED)
+                                        }
+                                    }
+                                    _ => {
+                                        // all else is sent to the router
                                         let _ = service
                                             .router_send
                                             .send(RouterMessage::PubsubMessage(id, source, message))
                                             .map_err(|_| {
                                                 debug!(service.log, "Failed to send pubsub message to router");
                                             });
-                                    } else {
-                                        metrics::inc_counter(&metrics::GOSSIP_UNAGGREGATED_ATTESTATIONS_IGNORED)
                                     }
                                 }
-                                _ => {
-                                    // all else is sent to the router
-                                    let _ = service
-                                        .router_send
-                                        .send(RouterMessage::PubsubMessage(id, source, message))
-                                        .map_err(|_| {
-                                            debug!(service.log, "Failed to send pubsub message to router");
-                                        });
-                                }
                             }
+                            // Then why do we send it?
+                            BehaviourEvent::PeerSubscribed(_, _) => {},
```

Comment:
This is used in gossipsub tests. We don't use it higher up. Its likely the gossipsub tests may need to be reworked